content/en/docs/tutorials/configuration/updating-configuration-via-a-configmap.md

```diff
@@ -0,0 +1,675 @@
+```
+Found 3 pods, using pod/configmap-env-var-6d9ff89fb6-bzcf6
+Thu Jan  4 16:30:35 UTC 2024 The basket is full of mangoes
+Thu Jan  4 16:30:45 UTC 2024 The basket is full of mangoes
+Thu Jan  4 16:30:55 UTC 2024 The basket is full of mangoes
+```
+
+This demonstrates the scenario of updating environment variables in a Pod that are derived
+from a ConfigMap. Changes to the ConfigMap values are applied to the Pod during the subsequent
+rollout. If Pods get created for another reason, such as scaling up the Deployment, then the new Pods
+also use the latest configuration values; if you don't trigger a rollout, then you might find that your
+app is running with a mix of old and new environment variable values.
+
+
+## Update configuration via a ConfigMap in a multi-container Pod {#rollout-configmap-multiple-containers}
+
+Use the `kubectl create configmap` command to create a ConfigMap from [literal values](/docs/tasks/configure-pod-container/configure-pod-configmap/#create-configmaps-from-literal-values):
+
+```shell
+kubectl create configmap color --from-literal=color=red
+```
+Below is an example manifest for a Deployment that manages a set of Pods, each with two containers. The two containers share an `emptyDir` volume that they use to communicate.
+The first container runs a webserver (`nginx`). The mount path for the shared volume in the webserver container is `/usr/share/nginx/html`. The second helper container is based on `alpine`, and for this container the `emptyDir` volume is mounted at `/pod-data`. The helper container writes a file in HTML that has its content based on a ConfigMap. The webserver container serves the HTML via HTTP.
+
+{{% code_sample file="deployments/deployment-with-configmap-two-containers.yaml" %}}
+
+Create the Deployment:
+```shell
+kubectl apply -f https://k8s.io/examples/deployments/deployment-with-configmap-two-containers.yaml
+```
+
+Check the pods for this Deployment to ensure they are ready (matching by
+{{< glossary_tooltip text="selector" term_id="selector" >}}):
+```shell
+kubectl get pods --selector=app.kubernetes.io/name=configmap-two-containers
+```
+You should see an output similar to:
+```
+NAME                                        READY   STATUS    RESTARTS   AGE
+configmap-two-containers-565fb6d4f4-2xhxf   2/2     Running   0          20s
+configmap-two-containers-565fb6d4f4-g5v4j   2/2     Running   0          20s
+configmap-two-containers-565fb6d4f4-mzsmf   2/2     Running   0          20s
+````
+
+Expose the Deployment (the `kubectl` tool creates a
+{{<glossary_tooltip text="Service" term_id="service">}}  for you):
+```shell
+kubectl expose deployment configmap-two-containers --name=configmap-service --port=8080 --target-port=80
+```
+
+Use ```kubectl``` to forward the port:
+```shell
+kubectl port-forward service/configmap-service 8080:8080 & # this stays running in the background
+```
+
+Access the service.
+```shell
+curl http://localhost:8080
+```
+
```

Comment:
Is this Service creation relevant to the topic?

Reply:
It lets people see the impact of the ConfigMap change, right?

Reply:
Yes, the service creation is relevant. The section on multi-container pods is serving an html page on nginx. To enable http requests from outside the cluster we're creating a ClusterIP service and using the kubectl port-forward functionality to reach it. On minikube, I prefer using the port-forwarding than creating a NodePort service.